utils/merkle-tree/src/lib.rs

```diff
@@ -30,40 +30,73 @@
 //! does not perform any sorting of the input data (leaves) nor when inner nodes are created.
 //!
 //! If the number of leaves is not even, last leaf (hash of) is promoted to the upper layer.
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+#[cfg(not(feature = "std"))]
+use alloc::vec;
+#[cfg(not(feature = "std"))]
+use alloc::vec::Vec;
+
+/// Supported hashing output size.
+///
+/// The size is restricted to 32 bytes to allow for a more optimised implementation.
+pub type Hash = [u8; 32];
+use hash_db::Hasher;
+
+mod keccak256 {
+	use tiny_keccak::{Hasher as _, Keccak};
+	/// Keccak256 hasher implementation.
+	pub struct Keccak256;
+	impl Keccak256 {
+		/// Hash given data.
+		pub fn hash(data: &[u8]) -> super::Hash {
+			<Keccak256 as super::Hasher>::hash(data)
+		}
+	}
```

Comment:
In general you seem to try to make this as generic as possible, but then you add this hasher here? I would remove this. This isn't that much code, that you can just reimplement where you need it. For beefy we can put it into the beefy primitive crate.

Reply:
have remove `Keccak256` module.